### PR TITLE
Hetzner dedicated server firewall module

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Every (?) dedicated server at Hetzner comes with a simple stateless [firewall](https://wiki.hetzner.de/index.php/Robot_Firewall/en) which can be configured either via the Robot UI, or via a REST API. This module allows to configure it via the [REST API](https://robot.your-server.de/doc/webservice/en.html#get-firewall-server-ip).

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
hetzner_firewall

### CODE REVIEW

**Comment by felixfontein** [comment]
CC @resmo @cschmitt-hcloud @LKaemmerling

File: lib/ansible/modules/net_tools/hetzner_firewall.py [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+  timeout:
+    description:
+      - Timeout (in seconds) for waiting for firewall to be configured.
+    type: int
+    default: 180
+'''
+
+EXAMPLES = r'''
+- name: Configure firewall for server with main IP 1.2.3.4
+  hetzner_failover_ip:
```
Comment (resmo) [inline lib/ansible/modules/net_tools/hetzner_firewall.py]: this should be `hetzner_firewall` right?

File: lib/ansible/modules/net_tools/hetzner_firewall.py [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+    type: str
+    choices: [ main, kvm ]
+    default: main
+  state:
+    description:
+      - Status of the firewall.
+      - Firewall is active if state is C(present), and disabled if state is C(absent).
+    required: yes
+    type: str
+    choices: [ present, absent ]
```
Comment (resmo) [inline lib/ansible/modules/net_tools/hetzner_firewall.py:52]: in many modules state is defaulted to `present`. While not a blocker, I would apply this default here as well.

**Review by resmo — COMMENTED** [review]
See comments inline, code LGTM.

File: lib/ansible/modules/net_tools/hetzner_firewall.py (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/modules/net_tools/hetzner_firewall.py]: Ah yes. Looks like I forgot this "tiny detail" while adjusting the example... :)

File: lib/ansible/modules/net_tools/hetzner_firewall.py (hunk 2 above)
Comment (felixfontein) [reply lib/ansible/modules/net_tools/hetzner_firewall.py:52]: Good point. Also, `present` is the recommended state IMO, so having it as a default is fine :)

**Review by LKaemmerling — APPROVED** [review]
LGTM

**Review by resmo — APPROVED** [review]
shipit

**Comment by felixfontein** [comment]
@resmo @LKaemmerling thanks a lot for reviewing! :)

**Pull request merged**